src/Analyser/MutatingScope.php

```diff
@@ -1969,16 +2150,25 @@ private function resolveType(Expr $node): Type
 						new ConstantStringType(';'),
 					]);
 				}
-				if ($resolvedConstantName === 'PHP_EOL') {
-					return new UnionType([
-						new ConstantStringType("\n"),
-						new ConstantStringType("\r\n"),
+				// iconv, https://www.php.net/manual/en/iconv.constants.php
+				if ($resolvedConstantName === 'ICONV_IMPL') {
+					return new IntersectionType([
+						new StringType(),
+						new AccessoryNonEmptyStringType(),
 					]);
 				}
-				if ($resolvedConstantName === '__COMPILER_HALT_OFFSET__') {
-					return new IntegerType();
+				// libxml, https://www.php.net/manual/en/libxml.constants.php
+				if ($resolvedConstantName === 'LIBXML_VERSION') {
+					return IntegerRangeType::fromInterval(1, null);
 				}
-				if ($resolvedConstantName === 'PHP_INT_MAX') {
+				if ($resolvedConstantName === 'LIBXML_DOTTED_VERSION') {
+					return new IntersectionType([
+						new StringType(),
+						new AccessoryNonEmptyStringType(),
+					]);
+				}
+				// openssl, https://www.php.net/manual/en/openssl.constants.php
+				if ($resolvedConstantName === 'OPENSSL_VERSION_NUMBER') {
 					return IntegerRangeType::fromInterval(1, null);
 				}
```

Comment:
would it make sense to also add a `E_*` constants? would this be usefull for bitmask-logic bugs?

Reply:
Looks like phpstan can resolve them already via reflection or so. But I added tests for them as well